packages/esm-patient-conditions-app/src/conditions/conditions-form.component.tsx

```diff
@@ -1,51 +1,103 @@
-import React, { SyntheticEvent, useMemo } from 'react';
+import React, { SyntheticEvent, useCallback, useEffect, useMemo, useState } from 'react';
 import 'dayjs/plugin/utc';
 import { useTranslation } from 'react-i18next';
 import { BehaviorSubject } from 'rxjs';
-import { Button, ButtonSet, Form } from '@carbon/react';
+import { Button, ButtonSet, Form, InlineLoading, InlineNotification } from '@carbon/react';
 import { useLayoutType } from '@openmrs/esm-framework';
+import { ConditionDataTableRow } from './conditions.resource';
 import ConditionsWidget from './conditions-widget.component';
 import styles from './conditions-form.scss';
-import { ConditionDataTableRow } from './conditions.resource';
 
 interface ConditionFormProps {
+  closeWorkspace: () => void;
   condition?: ConditionDataTableRow;
+  formContext: 'creating' | 'editing';
   patientUuid?: string;
-  context: string;
-  closeWorkspace: () => void;
 }
 
-const ConditionsForm: React.FC<ConditionFormProps> = ({ closeWorkspace, patientUuid, context, condition }) => {
+const ConditionsForm: React.FC<ConditionFormProps> = ({ closeWorkspace, condition, formContext, patientUuid }) => {
   const { t } = useTranslation();
   const isTablet = useLayoutType() === 'tablet';
-  const [hasSubmissibleValue, setHasSubmissibleValue] = React.useState(false);
   const submissionNotifier = useMemo(() => new BehaviorSubject<{ isSubmitting: boolean }>({ isSubmitting: false }), []);
```

Comment:
Do we need the submissionNotifier at all? I.e., can't we just use `isSubmittingForm` / `setIsSubmittingForm` directly?

Reply:
@ibacher There was this idea of supporting "Forms Embeddable widgets". This was in effort to reuse existing RefApp widgets within Forms as extensions. This works on a concept of "self contained components" and stuff like the `submissionNotifier` helped facilitate that.  Checkout thread for some background info => https://talk.openmrs.org/t/o3-embedding-conditions-widget-in-forms/36642

Reply:
... I don't love that, but ok.